File: A2/earth_electrons.cpp
// Copyright 2017 Sihan Wang shwang95@bu.edu

#include <iostream>
#include <cmath>
int main() {
	double earth_mass = 5.972e27;
	double mol_mass = 64;
	double avg_mol = 6.023e23;
	double elec_num = 8;
	double num = earth_mass / mol_mass;
	double elect_num = num * elec_num;
	double tb = pow(2, 40);
	double storage = elect_num * avg_mol / tb;
	double storage_min = storage * 0.9;
	double storage_max = storage * 1.1;
	std::cout << storage << std::endl;
	std::cout << storage_min << std::endl;
	std::cout << storage_max << std::endl;
}

File: A2/earth_electrons.py
# Copyright 2017 Sihan Wang shwang95@bu.edu

earth_mass = 5.972e27
mol_mass = 65
avg_mol = 6.023e23
elec_num = 8
num = earth_mass / mol_mass
elect_num = num * elec_num
tb = 2 ** 40
storage = elect_num * avg_mol / tb
storage_min = storage * 0.9
storage_max = storage * 1.1
print("%e"%storage)
print("%e"%storage_min)
print("%e"%storage_max)